api/src/main/java/org/apache/iceberg/catalog/TableIdentifier.java

```diff
@@ -45,12 +46,14 @@ public static TableIdentifier of(Namespace namespace, String name) {
   }
 
   public static TableIdentifier parse(String identifier) {
+    Preconditions.checkArgument(identifier != null, "Table identifier must be non-null");
     Iterable<String> parts = DOT.split(identifier);
     return TableIdentifier.of(Iterables.toArray(parts, String.class));
   }
 
   private TableIdentifier(Namespace namespace, String name) {
-    Preconditions.checkArgument(name != null && !name.isEmpty(), "Invalid table name %s", name);
+    Preconditions.checkArgument(namespace != null, "Namespace must be non-null");
+    Preconditions.checkArgument(name != null && !name.isEmpty(), "Table name must be non-null/non-empty");
```

Comment:
There's no need to move this check. Reordering without a clear reason just results in more commit conflicts later.

Also, the original error message conforms to our conventions better than the new one. Our convention is to be as direct as possible, which is why it started with "Invalid table name". If you'd like to add the "non-null/non-empty" part, I'd recommend using `"Invalid table name (null or empty): %s"`.

Reply:
sure thing. I would probably just have `Invalid table name (null or empty)` without printing what the actual `name` is, since it's already mentioned in the error message `(null or empty)`

Reply:
It's a minor distinction, but I would normally include what the user actually had. If it was null, then it will show up as `Invalid table name (null or empty): null`. Not a big deal the way it is here, though.